x/mock/simulation/util.go

```diff
@@ -8,9 +8,26 @@ import (
 	"testing"
 	"time"
 
+	"github.com/cosmos/cosmos-sdk/baseapp"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/tendermint/abci/types"
 )
 
+// assertAll asserts the all invariants against application state
+func assertAllInvariants(t *testing.T, app *baseapp.BaseApp, invs sdk.Invariants,
+	event string, displayLogs func()) {
+
+	ctx := app.NewContext(false, abci.Header{Height: app.LastBlockHeight() + 1})
+
+	for i := 0; i < len(invs); i++ {
+		if err := invs[i](ctx); err != nil {
+			fmt.Printf("Invariants broken after %s\n%s\n", event, err.Error())
```

Comment:
Would be nice if it checked all the invariants here and returned all the errors, instead of just exiting on the first one.

Reply:
this is just copied code, opening an issue: https://github.com/cosmos/cosmos-sdk/issues/3636